p24/tests/unit_tests.rs

```diff
@@ -0,0 +1,33 @@
+// use p24::refs;
+
+#[test]
+
+pub fn test_f1() {
```

Comment:
You implemented integration tests, the slides asked to implement unit tests instead. In other words, these tests should reside in the refs module.